models/spectrogram_cnn.py

```diff
@@ -61,70 +65,89 @@ def assemble_model(src: np.ndarray,
                                 activation=arch_layer.activation,
                                 data_format=data_format,)(x)
 
+    # Flatten down to a single dimension
+    x = keras.layers.Flatten()(x)
+
     # @paper: sigmoid activations with binary cross entropy loss
     # @paper: FC-512
     x = keras.layers.Dense(512)(x)
 
     # @paper: FC-368(sigmoid)
     outputs = keras.layers.Dense(
-        368, activation='sigmoid', name='predictions')(x)
+        n_outputs, activation='sigmoid', name='predictions')(x)
 
     return keras.Model(inputs=inputs, outputs=outputs)
 
-
 if __name__ == "__main__":
-    # Load audio sample
-    input_audio_path: str = os.getenv('AUDIO_WAV_INPUT')
-    # Define audio sample max duration
-    duration: float = 1
-    # Extract raw audio
-    y_audio, sample_rate = input_raw_audio(input_audio_path, duration=duration)
+
+
+    # TEMP!
+
+    dataset: str = os.getcwd() + "/" + os.getenv('TRAINING_SET')
+    params = {
+            'data_file':dataset,
+            'batch_size': 64,
+            'shuffle': True
+            }
+
+    training_generator = SoundDataGenerator(first=0.8, **params)
+    validation_generator = SoundDataGenerator(last=0.2, **params)
+
+    n_samples = training_generator.get_audio_length()
+    n_outputs = training_generator.get_label_size()
```

Comment:
@mo-seph I wonder if the size of the output expected would be 368 and not 16? There's one step mentioned in the generation step where they mention concatenating all the vectors into a single super-vector (368) before outputting the audio signal (16384).  What do you think?